### PR TITLE
feat(cdk): add byPropertyValue utility

### PR DESCRIPTION
**Pull request by nikitasadchikov**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/cdk/array/src/by-property-value.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export function byPropertyValue<ArrayEntryType, ArrayEntryTypeKey extends keyof ArrayEntryType>(
```
Comment (splincode) [inline packages/cdk/array/src/by-property-value.ts:1]: prefer for DX
`byPropertyValue('a', 'title')` -> `findByWhere({ key: 'a', value: 'title' })`

Comment (Schirbak) [reply packages/cdk/array/src/by-property-value.ts:1]: you are sure?

```ts
[{ id: 1, draft: true }, { id: 2, draft: false }].find(byPropertyValue('draft', true)) 
[{ id: 1, draft: true }, { id: 2, draft: false }].filter(byPropertyValue('draft', true)) 
```
vs
```ts
[{ id: 1, draft: true }, { id: 2, draft: false }].find(findByWhere({ key: 'draft', value: true })) 
[{ id: 1, draft: true }, { id: 2, draft: false }].filter(findByWhere({ key: 'draft', value: true })) 
```

Comment (Schirbak) [reply packages/cdk/array/src/by-property-value.ts:1]: funny semblance of jest:
```ts
const notes = [{ id: 1, draft: true }, { id: 2, draft: false }];
notes.find(whereProperty('draft').is(true))
notes.find(whereProperty('draft').isTrue()) 
notes.find(whereProperty('id').isNot(2)) 
notes.find(whereProperty('id').isIn([1, 2])) 
notes.find(whereProperty('id').isNotIn([1, 2]))

Comment (Schirbak) [reply packages/cdk/array/src/by-property-value.ts:1]:
```ts
notes.find(where(property('id').isBetween(1, 2) |> and |> property('draft').isTrue()))) 
```
:peka:

Comment (splincode) [reply packages/cdk/array/src/by-property-value.ts:1]: sounds good with your examples

Comment (splincode) [reply packages/cdk/array/src/by-property-value.ts:1]: maybe, can we use a jest utils style for filter behavior?

Comment (Schirbak) [reply packages/cdk/array/src/by-property-value.ts:1]: i mean, we won, but the solution is too complex

Comment (splincode) [reply packages/cdk/array/src/by-property-value.ts:1]: Okey, not problem

also next PR you can add `whereProperty` from your playground for as an alternative function `byPropertyValue`

File: packages/cdk/array/src/by-property-value.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    key: ArrayEntryTypeKey,
+    value: ArrayEntryType[ArrayEntryTypeKey]
+): (arrayItem: ArrayEntryType) => boolean {
```
Comment (Schirbak) [inline packages/cdk/array/src/by-property-value.ts:4]:
```ts
arrayItem is ArrayEntryType & { [key in ArrayEntryTypeKey]: ValueType }
```

**Review by Schirbak — APPROVED** [review]

**Pull request merged**